gflownet.py

```diff
@@ -1238,13 +1262,21 @@ def logq(path_list, actions_list, model, env):
     # TODO: this method is probably suboptimal, since it may repeat forward calls for
     # the same nodes.
     log_q = torch.tensor(1.0)
+    loginf = 1e3
     for path, actions in zip(path_list, actions_list):
         path = path[::-1]
         actions = actions[::-1]
         path_obs = np.asarray([env.state2obs(state) for state in path])
+        done = [0] * len(path)
+        masks = tf(
+            [
+                env.get_mask_invalid_actions(path[idx], done[idx])
+                for idx in range(len(path))
+            ]
+        )
         with torch.no_grad():
-            # TODO: potentially mask invalid actions next_q
             logits_path = model(tf(path_obs))
+        logits_path = torch.where(masks == 0, logits_path, -loginf)
```

Comment:
Would it not be `masks == 1`? We want to make `-loginf` the invalid (mask == True) actions.

Reply:
Sorry, got `torch.where` wrong. All good!